### PR TITLE
Merge main candidate from 09/18/2020

### PR DESCRIPTION
**Pull request by gustavo-marques**
Merging the main candidate from GFDL:

https://github.com/NOAA-GFDL/MOM6/pull/1211

There has been some reorganization of the source code and, therefore, changes in MOM_interface are needed to compile CESM/MOM6 from now on.  In addition, the default value for many parameters has been changed. Although we foresaw these changes and already modified the default parameters used in CESM/MOM6 accordingly (https://github.com/ESCOMP/MOM_interface/pull/65), care must be taken before this PR is merged to make sure answers in aux_mom are not changed. 

 I have confirmed that this PR does not change the answers in MOM6-examples using intel. It remains to be tested whether we get b4b for the aux_mom tests.

### CODE REVIEW

**Comment by alperaltuntas** [comment]
@gustavo-marques we are getting answer changes in all our tests except BMOM.
test directory: /glade/scratch/altuntas/cesm.tests/cheyenne/aux_mom/201015

**Comment by gustavo-marques** [comment]
Thanks, @alperaltuntas. This is likely due to changes in the default parameters. I will look into it this afternoon.

**Comment by alperaltuntas** [comment]
@gustavo-marques , two differences between our GMOM and BMOM are the following flags:
 RESTORE_SALINITY
 ADJUST_NET_FRESH_WATER_TO_ZERO

At least one of these must be causing forced cases to have answer changes.

File: config_src/mct_driver/mom_surface_forcing_mct.F90 [hunk 1]
```diff
@@ -1095,7 +1095,7 @@ subroutine surface_forcing_init(Time, G, US, param_file, diag, CS, restore_salt,
     call get_param(param_file, mdl, "USE_NET_FW_ADJUSTMENT_SIGN_BUG", &
                  CS%use_net_FW_adjustment_sign_bug, &
                    "If true, use the wrong sign for the adjustment to "//&
-                   "the net fresh-water.", default=.false.)
+                   "the net fresh-water.", default=.true.)
```
Comment (gustavo-marques) [inline config_src/mct_driver/mom_surface_forcing_mct.F90]: The default for this parameter did not have to be changed. We (NCAR) had already set this to False and, unintentionally, @Hallberg-NOAA set it back to True in https://github.com/NOAA-GFDL/MOM6/pull/1127

File: config_src/nuopc_driver/mom_surface_forcing_nuopc.F90 [hunk 2]
```diff
@@ -1094,7 +1094,7 @@ subroutine surface_forcing_init(Time, G, US, param_file, diag, CS, restore_salt,
     call get_param(param_file, mdl, "USE_NET_FW_ADJUSTMENT_SIGN_BUG", &
                  CS%use_net_FW_adjustment_sign_bug, &
                    "If true, use the wrong sign for the adjustment to "//&
-                   "the net fresh-water.", default=.false.)
+                   "the net fresh-water.", default=.true.)
```
Comment (gustavo-marques) [inline config_src/nuopc_driver/mom_surface_forcing_nuopc.F90]: Same here. The default should be False.

**Comment by gustavo-marques** [comment]
After manually setting ``USE_NET_FW_ADJUSTMENT_SIGN_BUG=False``, I was able to get b4b with dev/ncar. I will suggest that the default for this flag should be changed to false in the PR from GFDL, so let's wait for this patch before merging this PR.

I also noticed that changes in MOM_hor_visc.F90 introduced in https://github.com/NCAR/MOM6/pull/158 lead to changes in answers since we have LEITH_AH = True (@alperaltuntas mentioned that when merging https://github.com/NCAR/MOM6/pull/158).

**Comment by gustavo-marques** [comment]
@alperaltuntas, this PR is ready to be merged.

**Review by alperaltuntas — APPROVED** [review]

**Pull request merged**